tools/donate_cpu_lib.py

```diff
@@ -325,15 +325,15 @@ def __run_command(cmd, print_cmd=True):
     return return_code, stdout, stderr, elapsed_time
 
 
-def scan_package(work_path, cppcheck_path, jobs, libraries, capture_callstack=True):
+def scan_package(work_path, cppcheck_path, source_path, jobs, libraries, capture_callstack=True):
     print('Analyze..')
     os.chdir(work_path)
     libs = ''
     for library in libraries:
         if os.path.exists(os.path.join(cppcheck_path, 'cfg', library + '.cfg')):
             libs += '--library=' + library + ' '
 
-    dir_to_scan = 'temp'
+    dir_to_scan = os.path.basename(source_path)
```

Comment:
This assumes `source_path` is a subfolder of `work_path` - which currently is true but might not be in the future.

I will prepare another PR which fixes this as well as removing all remaining `os.chdir()` usages.